### ISSUE TITLE
Executable not readable or lacks permissions for Laravel Pint.

### ISSUE THREAD

**Issue by FrancisMawn** [open-southeners/vscode-laravel-pint#32]
**Describe the bug**
Got this error in the console (wasn't there before): 
["INFO" - 7:36:37 a.m.] Extension Name: open-southeners.laravel-pint.
["INFO" - 7:36:37 a.m.] Extension Version: 1.1.4.
["ERROR" - 7:36:52 a.m.] Executable not readable or lacks permissions for Laravel Pint.

**Environment:**
 - Operating system: Linux
 - IDE / version: VSCode 1.73.1
 - Extension version 1.1.4
 - Extension's config: 
 ```json
{
  "laravel-pint.executablePath": "./code/vendor/bin/pint",
  "laravel-pint.preset": "psr12",
  "laravel-pint.configPath": "./code/pint.json"
}
```

**Comment by d8vjork** [comment]
Ok, could be a regression introduced in #30 

Will take a look once I got time

**Comment by bilogic** [comment]
![image](https://user-images.githubusercontent.com/946010/219574367-74405f5d-3d5e-4a6d-87fd-e3603bbcb591.png)

I specified a pint path with global executable permissions, but also getting this same error

**Comment by rajeshtva** [comment]
"laravel/pint": `"^1.7"` 
extension version: `v1.1.4`

i noticed that this extension formats `.php` files properly but this extension also tries to format other file formats like `.vue` and `.js` & `.ts` files.

**Comment by rajeshtva** [comment]
@d8vjork my well wishes.

```
["INFO" - 21:20:15] Extension Name: open-southeners.laravel-pint.
["INFO" - 21:20:15] Extension Version: 1.1.4.
```

these lines appear when i try to format `.vue` file
```
["ERROR" - 21:20:17] Executable not readable or lacks permissions for Laravel Pint.
["ERROR" - 21:20:31] Executable not readable or lacks permissions for Laravel Pint.
```

but these lines appear when i try to format `.php` files
["INFO" - 21:20:48] Formatting completed in 444ms.
["INFO" - 21:20:49] Formatting completed in 350ms.

my extension settings: 
```json
{
 "[php]": {
    "editor.defaultFormatter": "open-southeners.laravel-pint"
  },
  "laravel-pint.enable": true,
  "laravel-pint.preset": "laravel",
}
```

these are all settings i could find in `~/.config/Code/User/settings.json`,

**Comment by bilogic** [comment]
Still getting this in 1.1.4

```
["INFO" - 7:46:15 PM] Extension Name: open-southeners.laravel-pint.
["INFO" - 7:46:15 PM] Extension Version: 1.1.4.
["ERROR" - 7:46:30 PM] Executable not readable or lacks permissions for Laravel Pint.
["ERROR" - 7:46:35 PM] Executable not readable or lacks permissions for Laravel Pint.
["ERROR" - 7:47:22 PM] Executable not readable or lacks permissions for Laravel Pint.
```

**Comment by d8vjork** [comment]
@rajeshtva There must be something wrong, I tested the extension on L9 and L10, didn't manage to get those messages, either way Laravel version doesn't matter for Pint

**Comment by khepin** [comment]
Had the same issue. Restarted my laptop, reinstalled the extension and now ... it works....
Not something I'd want to have to do every day or even every week in the middle of work though.

Enabling "more debug logs" didn't show anything so ... don't have much more info.

**Comment by olivsinz** [comment]
> 4\. vendor/bin/pint

use `vendor/bin/pint`

it works for me

**Comment by ChaerilM** [comment]
i could run it with `php vendor/bin/pint` but not `vendor/bin/pint`

**Comment by Sophist-UK** [comment]
On Nov 25, 2022 @d8vjork commented:
> Ok, could be a regression introduced in #30
> 
> Will take a look once I got time

That was literally over a year ago and this issue which breaks the extension completely has still not been resolved. This extension needs to be marked as abandoned by Microsoft. I will uninstall and install an alternative.

**Comment by d8vjork** [comment]
@Sophist-UK This us why is an open source extension, completely free and open to everyone to send fixes to it, also it is a pre-release version, marked as it because of a reason

**Comment by d8vjork** [comment]
This must be resolved within v1.1.6 release as plenty of other issues

Closing this, if anyone had this same issue please comment here or reopen

**Comment by Polfo** [comment]
I had the  "Executable not readable or lacks permissions for Laravel Pint." because I installed pint on one project, but not the other (where the error occured). Solved by using  `composer require laravel/pint --dev`
